Sources_v3/StreamChatUI/ChatChannelList/ChatChannelView.swift

```diff
@@ -0,0 +1,135 @@
+//
+// Copyright © 2020 Stream.io Inc. All rights reserved.
+//
+
+import StreamChat
+import UIKit
+
+open class ChatChannelView<ExtraData: ExtraDataTypes>: UIView where ExtraData.Channel: NameAndImageProviding {
+    
+    // MARK: - Properties
+    
+    public let uiConfig: UIConfig<ExtraData>
+    
+    public var channel: _ChatChannel<ExtraData>? {
+        didSet { updateAppearance() }
+    }
```

Comment:
Hmm, I've never thought about setting up the data like this. I always thought about having something like `load(_ data: ChatChannel<ExtraData>)`. Why do you think this is better?

Reply:
Having a model stored in a property allows to access it any time, event from the outside, if that's needed.

Reply:
Hmm, OK, why not 🤷‍♂️ In that case, `AvatarView` should have `user: _ChatUser<ExtraData>?`, right?

Reply:
The idea was to have simple base `AvatarView` with an image and introduce `Channel/User/AvatarView: AvatarView` that own **channel/user** models later.

`AvatarView` is added in d53402bb0a05484f8f7b2dce2e1c3b55278815a2